src/iosrv/srv.c

```diff
@@ -201,6 +201,18 @@ normal_unit_pop(ABT_pool *pools, ABT_pool *pool)
 	return ABT_UNIT_NULL;
 }
 
+static unsigned int g_seed;
+inline void fast_srand(int seed)
+{
+	g_seed = seed;
+}
+
+inline int fast_rand(void)
+{
+	g_seed = (214013 * g_seed + 2531011);
+	return (g_seed >> 16) & 0x7FFF;
```

Comment:
where did you take that from? :)
i personally think that we should use the version provided by DPDK (https://doc.dpdk.org/api/rte__random_8h.html)

Reply:
Not sure why it impacts performance so much, I did run a simple benchmark, rand() can generate 160 million random numbers, this function can generate 300 million random numbers, it is faster but I think either of them should be fine giving the rate is 1000X faster than our IOPs/sec

Reply:
Ah, sure I will try to use rte_rand() then, I google it from an article in intel site.
I am trying to understand why rand generator impacts the performance this much. But simply adding an extra rand() into ds_sched_run() did create a lot overhead.